src/realm/group_shared.cpp

```diff
@@ -994,20 +994,16 @@ void SharedGroup::do_open(const std::string& path, bool no_create_file, bool is_
             bool file_format_ok = false;
             // In shared mode (Realm file opened via a SharedGroup instance) this
             // version of the core library is able to open Realms using file format
-            // versions 2, 3, 4, 5, 6, and 7. Version 2, 3, 4, 5, and 6 files need
-            // to be upgraded. Please see Allocator::get_file_format_version() for
-            // information about the individual file format verions.
+            // versions from 2 to Group::s_current_file_format_version. Please see
+            // Group::get_file_format_version() for information about the individual
+            // file format versions.
             switch (current_file_format_version) {
                 case 0:
                     file_format_ok = (top_ref == 0);
                     break;
-                case 2:
-                case 3:
-                case 4:
-                case 5:
-                case 6:
-                case 7:
-                    file_format_ok = true;
+                default:
+                    file_format_ok = (current_file_format_version >= 2) &&
+                                     (current_file_format_version <= Group::s_current_file_format_version);
```

Comment:
I don't like this mix of `switch` and "regular" logic. Not sure how to fix it in an elegant way, that doesn't require an update, when we bump the file format version.